R/backend-teradata.R

```diff
@@ -63,14 +63,19 @@ sql_translation.Teradata <- function(con) {
       bitwXor       = sql_prefix("BITXOR", 2),
       bitwShiftL    = sql_prefix("SHIFTLEFT", 2),
       bitwShiftR    = sql_prefix("SHIFTRIGHT", 2),
-      as.numeric    = sql_cast("NUMERIC"),
+      as.numeric    = function(x, digits = 9L) {
+                        sql_expr(CAST(!!x %as% DECIMAL(12L,9L)))
```

Comment:
I'm not sure if the `digits` argument makes sense (none of the other backends has one and R doesn't either).
In your current code it is not used anyway.